qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -137,99 +160,239 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
             Output dag with UnitaryGates synthesized to target basis.
 
         Raises:
-            TranspilerError:
-                1. pulse_optimize is True but pulse optimal decomposition is not known
-                   for requested basis.
-                2. pulse_optimize is True and natural_direction is True but a preferred
-                   gate direction can't be determined from the coupling map or the
-                   relative gate lengths.
+            TranspilerError: if a 'method' was specified for the class and is not
+                found in the installed plugins list. The list of installed
+                plugins can be queried with
+                :func:`~qiskit.transpiler.passes.synthesis.plugins.unitary_synthesis_plugin_names`
         """
+        if not self.method:
+            method = "default"
+        else:
+            method = self.method
+        if method not in self.plugins.ext_plugins:
+            raise TranspilerError("Specified method: %s not found in plugin list" % method)
+        default_method = self.plugins.ext_plugins["default"].obj
+        plugin_method = self.plugins.ext_plugins[method].obj
+        if plugin_method.supports_coupling_map:
+            dag_bit_indices = {bit: idx for idx, bit in enumerate(dag.qubits)}
+        kwargs = {}
+        if plugin_method.supports_basis_gates:
+            kwargs["basis_gates"] = self._basis_gates
+        if plugin_method.supports_natural_direction:
+            kwargs["natural_direction"] = self._natural_direction
+        if plugin_method.supports_pulse_optimize:
+            kwargs["pulse_optimize"] = self._pulse_optimize
+        if plugin_method.supports_gate_lengths:
+            kwargs["gate_lengths"] = _build_gate_lengths(self._backend_props)
+        if plugin_method.supports_gate_errors:
+            kwargs["gate_errors"] = _build_gate_errors(self._backend_props)
+        supported_basis = plugin_method.supported_basis
+        if supported_basis is not None:
+            kwargs["matched_basis"] = _choose_basis(self._basis_gates, supported_basis)
+
+        # Handle approximation degree as a special case for backwards compatibility, it's
+        # not part of the plugin interface and only something needed for the default
+        # pass.
+        default_method._approximation_degree = self._approximation_degree
+        if method == "default":
+            plugin_method._approximation_degree = self._approximation_degree
+
+        for node in dag.named_nodes(*self._synth_gates):
+            if plugin_method.supports_coupling_map:
+                kwargs["coupling_map"] = self._coupling_map
+                kwargs["qubits"] = [dag_bit_indices[x] for x in node.qargs]
```

Comment:
This _could_ be passed as a tuple to `coupling_map` if we're desperate to keep a one-to-one mapping between `supports_` and a kwarg.

I suppose it depends on if we're worried that we may also want to use the same kwarg `qubits` for another argument in the future, and the format might not be exactly the same.

Reply:
I'll make it a tuple just to reserve the potential in the future, there wasn't a particular reason I did it this way except for how I went about writing the option (I forgot about needing the operands at first and added it after I had written all the coupling_map stuff).

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/6124/commits/aa2af0fd0f84d0181118d433aa8b4fc77df8aac0